tools/clang/lib/SPIRV/SpirvBuilder.cpp

```diff
@@ -1100,6 +1117,24 @@ SpirvInstruction *SpirvBuilder::createSpirvIntrInstExt(
   return inst;
 }
 
+void SpirvBuilder::createBeginInvocationInterlockEXT(SourceLocation loc,
+                                                     SourceRange range) {
+  assert(insertPoint && "null insert point");
+
+  auto *inst = new (context)
+      SpirvNullaryOp(spv::Op::OpBeginInvocationInterlockEXT, loc, range);
```

Comment:
I requested switching to NullaryOp, since we already have Unary/BinaryOp for instructions that have no special handling at the class level, so that just seems more consistent to me than adding unique classes for every opcode if they have no unique behavior. If we have other cases that could be replaced I think that would be a fine followup, or fine to leave as-is.

Reply:
Makes sense, thanks for the context.